platforms/db2/src/main/java/io/ebean/platform/db2/BaseDB2Platform.java

```diff
@@ -53,4 +54,9 @@ public PlatformIdGenerator createSequenceIdGenerator(BackgroundExecutor be, Data
     return new DB2SequenceIdGenerator(be, ds, seqName, sequenceBatchSize);
   }
 
+  @Override
+  protected String withForUpdate(String sql, Query.LockWait lockWait, Query.LockType lockType) {
+    // NOWAIT and SKIP LOCKED not supported with Db2
+    return sql + " for update";
```

Comment:
@rob-bygrave 

I have another question about this PR.

The test TestQueryFindPagedList.test_forUpdate throws an exception because a query with join cannot be executed.

"For update" cannot be executed in these constellations: https://www.sqlerror.de/db2_sql_error_-511_sqlstate_42829.html

Would it be a viable way to check the SQL to see if any of the "forbidden" keywords appear in the command and then decide whether for update is possible? (e.g.: join, distinct, order by, group by, etc.) Or does it work if ebean passes on the exception to the DB (like the existing solution)?

Something like this:

```
protected String withForUpdate(String sql, Query.LockWait lockWait, Query.LockType lockType) {
    // NOWAIT and SKIP LOCKED not supported with Db2
    if (checkKeywords(sql)) {
        return sql + " for update";
    } else {
        return super.withForUpdate(sql, lockWait, lockType);
    }
}

private boolean checkKeywords(String sql) { ... }
```

Reply:
Thanks for the feedback, we don't have a usecase that forces to check the keywords. We are also agree with the merged solution.